ravenframework/Distributions.py

```diff
@@ -1656,13 +1658,20 @@ def _handleInput(self, paramInput):
         outcome = child.parameterValues["outcome"]
         value = child.value
         self.mapping[outcome] = value
-        if float(outcome) in self.values:
+        try:
+          float(outcome)
+          self.isFloat = True
+        except:
+          self.isFloat = False
+        if outcome in self.values:
           self.raiseAnError(IOError,'Categorical distribution has identical outcomes')
         else:
-          self.values.add(float(outcome))
+          self.values.add(float(outcome) if self.isFloat else outcome)
       else:
```

Comment:
@alfoa: This is a general comment, and do not need to address in this PR:

The assumption here is that the users can either provide float or string outcomes. If in the inputs, there are mixed floats and strings, the current implementation can not catch it, and an unexpected crash will happen. I would suggest add a check here to make sure the inputs are either all floats or all string values.